### PR TITLE
See-Through Paths (and implement gCurrentViewportFlags)

### PR DESCRIPTION
**Pull request by wolfreak99**
This adds the option to toggle see through for paths. This also implements the gCurrentViewportFlags variable as well, along with updating it from uint16 to uint32 to support more flags (which is needed for the seethrough_paths). Because supports are not yet completely native, I have kept it updagting the address portion as well (see viewport_paint for note), and to be safe, the flags are converted back to uint16 before being sent back to the address (though it seemed to work fine even as the address being a uint32, but i do not take part in the reverse engineering process so i can't give a concrete answer here)..
 
This is definitely not ready to be merged immediately as the above concern involving gCurrentViewportFlags needs to be addressed, but other than that, the see-through paths works great! It hides the paths, path items (i.e. benches, lamps, fountains, trashcans, etc), and also the path banners (not including the big scenery banners, only the banners that can be placed on paths and have the "no entry" functionality).

### CODE REVIEW

File: src/interface/viewport.h [hunk 1]
```diff
@@ -38,23 +38,24 @@ enum {
...
+	VIEWPORT_INTERACTION_ITEM_RIDE = 3,
+	VIEWPORT_INTERACTION_ITEM_WATER = 4,
+	VIEWPORT_INTERACTION_ITEM_SCENERY = 5,
+	VIEWPORT_INTERACTION_ITEM_FOOTPATH = 6,
+	VIEWPORT_INTERACTION_ITEM_FOOTPATH_ITEM = 7,
+	VIEWPORT_INTERACTION_ITEM_PARK = 8,
+	VIEWPORT_INTERACTION_ITEM_WALL = 9,
+	VIEWPORT_INTERACTION_ITEM_LARGE_SCENERY = 10,
+	VIEWPORT_INTERACTION_ITEM_LABEL = 11,
+	VIEWPORT_INTERACTION_ITEM_BANNER = 12,
```
Comment (IntelOrca) [inline src/interface/viewport.h]: Why have you added numbers?

**Comment by IntelOrca** [comment]
Apart from that, the rest of the code looks fine - I will test it out this evening.

File: src/paint/paint.c [hunk 2]
```diff
@@ -954,23 +963,35 @@ void paint_quadrant_ps() {
...
 					image_id &= 0x7FFFF;
 					image_id |= 0x41880000;
 				}
 			}
 		}
 		if (gCurrentViewportFlags & VIEWPORT_FLAG_SEETHROUGH_SCENERY) {
-			if (ps->sprite_type == 10 || ps->sprite_type == 12 || ps->sprite_type == 9 || ps->sprite_type == 5) {
+			if (ps->sprite_type == VIEWPORT_INTERACTION_ITEM_LARGE_SCENERY || 
+				ps->sprite_type == VIEWPORT_INTERACTION_ITEM_WALL || 
+				ps->sprite_type == VIEWPORT_INTERACTION_ITEM_SCENERY) {
```
Comment (marijnvdwerf) [inline src/paint/paint.c]: Why did you remove the check for `VIEWPORT_INTERACTION_ITEM_BANNER` (12)?

Comment (wolfreak99) [inline src/paint/paint.c]: I felt it would make more sense to have the banners be toggled with the see-through paths, it would also help finding hidden no-entry signs, say, on multiplayer or something, where they would normally be covered by other scenery. or you want to read the "TOMB RAIDERS AHEAD" path sign under the giant pyramid or something. It only effects the path banners and not the actual giant scenery banners.

File: src/interface/viewport.h (hunk 1 above)
Comment (wolfreak99) [inline src/interface/viewport.h]: @IntelOrca  Honestly, I don't remember why i put numbers here. I think it was when i was going through and comparing the sprite_types and didn't feel like recounting them out. Do you want me to go ahead and remove the numbers?

Comment (IntelOrca) [inline src/interface/viewport.h]: Yes please

Comment (duncanspumpkin) [inline src/interface/viewport.h]: It made sense in older versions of VS2015 to put in the numbers as it was annoying trying to double check the numbers. It now tells you when hovering over the value so it isn't really required. Thats one of the reasons I have sprinkled a few numbers in when implementing the track functions.

File: src/interface/viewport.c [hunk 3]
```diff
@@ -675,6 +676,10 @@ void viewport_render(rct_drawpixelinfo *dpi, rct_viewport *viewport, int left, i
  */
 void viewport_paint(rct_viewport* viewport, rct_drawpixelinfo* dpi, int left, int top, int right, int bottom){
 	gCurrentViewportFlags = viewport->flags;
+	//This should still be updated until the rest of supports, etc, is reverse-engineered. Otherwise
+	// Wooden Coaster supports (and a few other coaster supports) will still appear even if "invisible supports"
```
Comment (duncanspumpkin) [inline src/interface/viewport.c]: All coasters apart from jr rc. I wouldn't bother drawing attention to just wooden coasters just say unimplemented coasters.

File: src/interface/window.h [hunk 4]
```diff
@@ -89,7 +89,7 @@ typedef struct rct_viewport {
 	sint16 view_height;				// 0x0E
 	uint8 zoom;						// 0x10
 	uint8 var_11;
-	uint16 flags;					// 0x12
+	uint32 flags;					// 0x12
```
Comment (duncanspumpkin) [inline src/interface/window.h]: Update the structure size or remove the size ref. (See line 79)

File: src/interface/viewport.h (hunk 1 above)
Comment (IntelOrca) [inline src/interface/viewport.h]: Its useful for very big enums like ride or track type or when there are holes... but here there is no need.

**Comment by marijnvdwerf** [comment]
Instead of half-integrating it, what about just renaming `VIEWPORT_FLAG_15`? (Which doesn't seem to be used)

**Comment by wolfreak99** [comment]
@marijnvdwerf  I can see where you're coming from, but at the same time i honestly feel a little weird about doing that, what if later down the line flag_15 is used for something but hasn't been decompiled yet? maybe if we could get a second opinion if it's used or not i could consider it but at the current moment, every other view option is within gCurrentViewportFlags so it kinda just feels right. idk again this portion of stuff i am not advanced enough in so i just feel unsure on treading too deep in. i just like adding stuff where possible (i hope this makes sense)
